app-android/src/main/java/io/github/droidkaigi/confsched2023/license/OssLicenseRepositoryImpl.kt

```diff
@@ -0,0 +1,69 @@
+package io.github.droidkaigi.confsched2023.license
+
+import android.content.Context
+import io.github.droidkaigi.confsched2023.R.raw
+import io.github.droidkaigi.confsched2023.model.License
+import io.github.droidkaigi.confsched2023.model.OssLicenseRepository
+import kotlinx.collections.immutable.PersistentList
+import kotlinx.collections.immutable.persistentListOf
+import kotlinx.collections.immutable.toPersistentList
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import okio.BufferedSource
+import okio.buffer
+import okio.source
+import javax.inject.Inject
+
+class OssLicenseRepositoryImpl @Inject constructor(
```

Comment:
How about moving the logic to DefaultOssLicenseDataSource to simplify the app's implementation? I think this is similar to an API, since both handle data parsing. Therefore, I suggest transferring the code to DefaultOssLicenseDataSource.

```kotlin
class DefaultOssLicenseDataSource @Inject constructor(
    private val context: Context,
) : OssLicenseDataSource {

    override suspend fun licenseFlow(): OssLicense {
        return withContext(context = Dispatchers.IO) {
            val details = readLicensesFile().toRowList()
            val metadata = readLicensesMetaFile().toRowList().parseToLibraryItem(
                details = details,
            )
            val groupList = metadata.distinctBy { it.name }.groupByCategory()
                .map {
                    OssLicenseGroup(
                        title = it.key,
                        licenses = it.value,
                    )
                }.toPersistentList()
            OssLicense(groupList)
        }
    }

    private fun List<License>.groupByCategory(): Map<String, List<License>> {
        val categoryList = listOf(
            "Android Support",
            "Android Datastore",
            "Android ",
            "Compose UI",
            "Compose Material3",
            "Compose ",
            "AndroidX lifecycle",
            "AndroidX ",
            "Kotlin",
            "Dagger",
            "Firebase",
            "Ktorfit",
            "okhttp",
            "ktor",
        )
        return groupBy { license ->
            categoryList.firstOrNull {
                license.name.startsWith(
                    prefix = it,
                    ignoreCase = true,
                )
            } ?: "etc"
        }
    }

    private fun readLicensesMetaFile(): BufferedSource {
        return context.resources.openRawResource(raw.third_party_license_metadata)
            .source()
            .buffer()
    }

    private fun readLicensesFile(): BufferedSource {
        return context.resources.openRawResource(raw.third_party_licenses)
            .source()
            .buffer()
    }

    private fun List<String>.parseToLibraryItem(details: List<String>): List<License> {
        return mapIndexed { index, value ->
            val (position, name) = value.split(' ', limit = 2)
            val (offset, length) = position.split(':').map { it.toInt() }
            val id = name.replace(' ', '-')
            License(
                name = name,
                id = id,
                offset = offset,
                length = length,
                detail = details[index],
            )
        }
    }

    private fun BufferedSource.toRowList(): List<String> {
        val list: MutableList<String> = mutableListOf()
        while (true) {
            val line = readUtf8Line() ?: break
            list.add(line)
        }
        return list
    }
}
```